testing/architecture-test/build.gradle.kts

```
@@ -66,17 +86,19 @@ tasks {
         systemProperty("org.gradle.public.api.includes", (PublicApi.includes + PublicKotlinDslApi.includes).joinToString(":"))
         systemProperty("org.gradle.public.api.excludes", (PublicApi.excludes + PublicKotlinDslApi.excludes).joinToString(":"))
 
+        dependsOn(platformsDataResolvable, packageInfoDataResolvable)
+
         jvmArgumentProviders.add(
             ArchUnitPlatformsData(
                 layout.settingsDirectory.dir("platforms"),
-                rootProject.tasks.named("platformsData").get().outputs.files.elements.map { it.single() },
+                providers.provider { platformsDataResolvable.get().singleFile },
```

💅 Haven't tested it, but avoiding `provider {}` should help with automatic dependency inference:
```suggestion
                platformsDataResolvable.flatMap { it.elements }.map { it.single },
```

---

Alternatively, you could change ArchUnitPlatformsData's json field to a FileCollection and use `singleFile` inside asArguments. 

+1 we should try to avoid calling `provider {}`